src/common.speech/SynthesisAdapterBase.ts

```diff
@@ -216,26 +216,29 @@ export class SynthesisAdapterBase implements IDisposable {
         this.privErrorCallback = errorCallBack;
 
         this.privSynthesisTurn.startNewSynthesis(requestId, text, isSSML, audioDestination);
-        try {
-            const connection: IConnection = await this.fetchConnection();
-            await this.sendSynthesisContext(connection);
-            await this.sendSsmlMessage(connection, ssml, requestId);
-            const synthesisStartEventArgs: SpeechSynthesisEventArgs = new SpeechSynthesisEventArgs(
-                new SpeechSynthesisResult(
-                    requestId,
-                    ResultReason.SynthesizingAudioStarted,
-                )
-            );
-
-            if (!!this.privSpeechSynthesizer.synthesisStarted) {
-                this.privSpeechSynthesizer.synthesisStarted(this.privSpeechSynthesizer, synthesisStartEventArgs);
-            }
 
-            const messageRetrievalPromise = this.receiveMessage();
+        try {
+            await this.connectImpl();
         } catch (e) {
             this.cancelSynthesisLocal(CancellationReason.Error, CancellationErrorCode.ConnectionFailure, e);
             return Promise.reject(e);
         }
+
+        const connection: IConnection = await this.fetchConnection();
```

Comment:
So this line and the block below are no longer in a try/catch block. What happens if one of the lines throws an exception now?

Reply:
thx, I will move this into try/catch block.